### ISSUE TITLE
helm: broken ingress

### ISSUE THREAD

**Issue by wrenix** [matrix-org/dendrite#3175]
current in helm-chart version: 0.13.1 (after #3077)

i got the host `https` with the following values:
```yaml
dendrite_config:
  global:
    server_name: "example.net"
    # without https needed for https://federationtester.matrix.org
    well_known_server_name: "dendrite.example.net:443"
    # https needed for fluffychat
    well_known_client_name: "https://dendrite.example.net:443"
``` 

@Omar007 and @S7evinK do you take a look?

**Comment by Omar007** [comment]
I had a hard time understanding what you were saying but thanks to the config I'm suspecting specifically the client part of the ingress (ingress paths `/_matrix/client` and `/_matrix/media`) gets the host value `https` assigned instead of the expected `dendrite.example.net`?

Assuming that, I double checked the specs and I did indeed make a mistake here calculating the host values for the ingress object. The well_known_client_name expects a base URL, not just a hostname+port like the well_known_server_name; https://spec.matrix.org/v1.7/client-server-api/#getwell-knownmatrixclient
Additionally, I also realized that they would not currently support IPv6 addresses as a value. I'll have to push a fix to cover these cases later. Apologies for the inconvenience.
